src/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -434,7 +385,10 @@ public SAMRecordIterator assertSorted(final SortOrder sortOrder) {
             final Container container = ContainerIO.readContainerHeader(newIterator.getCramHeader().getVersion().major, seekableStream);
             seekableStream.seek(seekableStream.position() + container.containerByteSize);
             iterator = newIterator;
-            iterator.jumpWithinContainerToPos(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX, SAMRecord.NO_ALIGNMENT_START);
+            boolean atAlignments;
+            do {
+                atAlignments = iterator.jumpToAlignment(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX, SAMRecord.NO_ALIGNMENT_START);
```

Comment:
Method name here is a little strange, as `jumpToAlignment` implies that it will immediately position the iterator at the requested position, but instead it appears to be iterating repeatedly in order to reach the requested position. Recommend renaming to make this more clear.